src/helix/core.cljs

```diff
@@ -55,20 +58,30 @@
         native? (or (keyword? type)
                     (string? type)
                     (:native (meta type)))
+        has-props? ^boolean (or (map? ?p)
+                                (nil? ?p))
+        children* ^seq (if has-props?
+                         ?c
+                         args)
```

Comment:
what are the type hints here for? it's not clear to me if they're doing anything. also, in the past I've typically seen them attached to the binding in a let rather than the expr, e.g.

```suggestion
        ^boolean has-props? (or (map? ?p)
                                (nil? ?p))
        ^seq children* (if has-props?
                         ?c
                         args)
```